src/Lecture2.hs

```diff
@@ -70,7 +78,11 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [Int] -> (Maybe Int, [Int])
+removeAt n list
+  | n < 0           = (Nothing, list)
+  | n > length list = (Nothing, list)
```

Comment:
The `length` function hangs on infinite lists so this function doesn't work on them.